reddit/users/api/views.py

```diff
@@ -73,14 +68,26 @@ def retrieve(self, request, *args, **kwargs):
         return response
 
 
-class UserPostsView(generics.ListAPIView):
-    queryset = User.objects.all()
-    serializer_class = UserPostsSerializer
-
-    def list(self, request, *args, **kwargs):
-        queryset = self.get_queryset()
-        serializer = self.get_serializer(queryset, many=True)
-        response = Response(serializer.data)
-        return response
-
+class RequestForgotUsernameMailView(generics.CreateAPIView):
+    serializer_class = UserDetailsSerializer
 
+    def create(self, request, *args, **kwargs):
+        email = request.data.get('email')
+        user = User.objects.get(email=email)
```

Comment:
Why are you using .get() here? It could through an error if the email is not present in the db.